Modules/_decimal/libmpdec/bits.h

```diff
@@ -145,7 +145,7 @@ mpd_bsf(mpd_size_t a)
 }
 /* END ASM */
 
-#elif defined(MASM)
+#elif defined(MASM) || defined (_M_ARM)
 #include <intrin.h>
```

Comment:
I think this is not necessary; at least the intention of `libmpdec` is that -DASM is defined on the command line for the Unix build and `/DMASM` for the MSVC build.

That's how the upstream `libmpdec` does it:

```
!if "$(MACHINE)" == "x64"
CONFIG = /DCONFIG_64 /DMASM
MPD_HEADER = mpdecimal64vc.h
MPD_PREC = 19
MPD_DPREC = 38
OBJS = $(OBJS) vcdiv64.obj
!endif

```

... and also the CPython build:

```
<PreprocessorDefinitions>_CRT_SECURE_NO_WARNINGS;MASM;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```

Reply:
That makes sense I think.  Do you consider armasm.exe to be the same as masm.exe?